### PR TITLE
Add implementation of fourth-kind Chebyshev polynomial smoother

### PR DESCRIPTION
**Pull request by MalachiTimothyPhillips**
From https://arxiv.org/pdf/2202.08830.pdf.

I've found these polynomials to be beneficial in certain cases: https://arxiv.org/pdf/2210.03179.pdf.

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -75,8 +75,8 @@ end
 @testset "  ex111: Chebyshev example                   " begin
     include("../examples/ex111_chebyshev.jl")
 
-    @test minimum(eigenvalues) ≈ -0.272875736279538
-    @test maximum(eigenvalues) ≈ 0.26048711552603726
+    @test minimum(eigenvalues) ≈ -0.2301200211798214
+    @test maximum(eigenvalues) ≈ 0.1306886015364882
```
Comment (jeremylt) [inline test/runtests.jl:86]: I'd like to check that both the old behavior is retained and the new behavior works as intended here. Could we add a `ex112_chebyshev_fourth.jl` and keep this one as is?

File: examples/ex111_chebyshev.jl [hunk 2]
```diff
@@ -14,6 +14,7 @@ diffusion = GalleryOperator("diffusion", p + 1, p + 1, mesh)
 
 # Chebyshev smoother
 chebyshev = Chebyshev(diffusion)
+setchebyshevtype(chebyshev, ChebyshevType.fourth)
```
Comment (jeremylt) [inline examples/ex111_chebyshev.jl]: I'd prefer to use 
```julia
chebyshev.type = ChebyshevType.forth
```
here. You can modify `function Base.setproperty!(preconditioner::Chebyshev, f::Symbol, value)` to do that. Let me spin up an PR to do that for `seteigenvalueestimatescaling`

File: src/PC/Chebyshev.jl [hunk 3]
```diff
@@ -50,20 +52,30 @@ mutable struct Chebyshev <: AbstractPreconditioner
...
     eigenvaluebounds::Array{Float64,1}
+    chebyshevtype::ChebyshevType.ChebyType
 
     # data empty until assembled
     eigenvalueestimates::Array{Float64,1}
     operatordiagonalinverse::AbstractArray{Float64}
 
     # inner constructor
-    Chebyshev(operator::Operator) = new(operator, [0.0, 0.1, 0.0, 1.0])
+    Chebyshev(operator::Operator) = new(operator, [0.0, 0.1, 0.0, 1.0], ChebyshevType.first)
```
Comment (jeremylt) [inline src/PC/Chebyshev.jl]: I think that the constructor should optionally accept a type with the default value of `ChebyshevType.first`

**Review by jeremylt — COMMENTED** [review]
This looks pretty great. Just a couple of comments.

File: test/runtests.jl (hunk 1 above)
Comment (MalachiTimothyPhillips) [reply test/runtests.jl:86]: Sure -- done!

File: examples/ex111_chebyshev.jl (hunk 2 above)
Comment (MalachiTimothyPhillips) [reply examples/ex111_chebyshev.jl]: Looks like virtually no alteration was needed to support this -- I've gone ahead and removed `setchebyshevtype`. While the example now sets the type in the ctor,

```
chebyshev.type = ChebyshevType.fourth
```

works as well.

File: src/PC/Chebyshev.jl (hunk 3 above)
Comment (MalachiTimothyPhillips) [reply src/PC/Chebyshev.jl]: Done

**Review by jeremylt — APPROVED** [review]
LGTM!

**Pull request merged**